### DISCUSSION TITLE
create-your-own-website-part1/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#17] [Announcements]
# create-your-own-website-part1/

تا حالا چند نفر از دوستان بصورت حضوری یا مجازی ازم خواستند که بگم چطوری و با چه پلتفرمی سایتم رو درست کردم. دو تا از پیام‌ها رو اینجا می‌گذارم و سعی می‌کنم برای اون‌ها بنویسم.
پیام اول:
 با سلام و احترام ، به قول چت جی پی تی امیدوارم که این ایمیل در بهترین حال و سلامتی به دست شما برسد. من … هستم، دانشجوی ارشد عمران دانشگاه شهید بهشتی تهران و البته متولد سال 1367.

https://aprd.ir/create-your-own-website-part1/

**Comment by alidalvand** [comment]
وب سایت زیر هم ویدئوهای خوبی گذاشته البته به زبان اصلی و ساده نویسی نکرده

**Reply by pourmand1376** [reply]
سلام علی جان.

ممنون از معرفی‌ ویدئوها. دوره ارزشمندی هست. یه نگاه اولیه به محتواش انداختم و به‌نظرم میاد چند مرحله‌ از چیزی که من می‌خوام بگم جلوتره. 

به نظرم میاد این دوره برای کسی که می‌خواد صرفاً یک وب‌سایت با تم آماده راه‌اندازی کنه، کمی زیاد هست ولی برای افرادی که می‌خوان حرفه‌ای‌تر ادامه بدن و معماری Hugo رو درک کنند که حتی بتونند تم سفارشی هم درست کنند، بسیار مناسب هست.

**Comment by alidalvand** [comment]
https://git.ir/udemy-migrate-from-wordpress-to-hugo-step-by-step/

**Comment by hossein2j** [comment]
سلام امیرجان
اتفاقا من هم همین سوالا رو در مورد وبلاگت داشتم. ممنون که جواب دادی. 

با نظرت موافقم. آزادی و راحتی که آدم توی وبلاگ شخصی داره، در مقایسه با استفاده از سرویس دهنده‌‌های وبلاگی مثل ویرگول و بلاگفا خیلی بیشتره. یه جورایی مثل زندگی در خانه و مِلک شخصی در مقایسه با سکونت در بلوک‌های آپارتمانی میمونه
البته بنظرم ویرگول، بجز محتوا روی سئو هم خوب داره کار میکنه. یعنی محتوایی که مرتبط باشه رو بهتر گوگل پیدا میکنه. پس در کنار معایبیش این نکته مثبت رو داره واسه افرادی که میخوان بنویسن اما با سئو آشنایی ندارن. 

پیروز باشی 
Hossein-jafari.ir

**Comment by alidalvand** [comment]
نظرت راجع به Next.js چیه ؟

**Reply by pourmand1376** [reply]
اونم خوبه. باهاش کار نکردم اما تعریفش رو شنیدم. 

به نظرم مهم‌ترین قسمتش پیدا کردن یک قالب (تمپلت) آماده و سریع هست که ازت خوشش بیاد. من از hugo خوشم اومد چون یک تمپلت خیلی زیبا پیدا کردم. 

وگرنه اگر قرار باشه بشینیم خودمون سایت رو با اون پلتفرم طراحی کنیم که اصلا نقض غرض میشه.

**Comment by alidalvand** [comment]
ما کماکان منتظریم :)

**Reply by pourmand1376** [reply]
سلام علی جان. 

احتمالا برای این مورد یک ویدئو درست کنم و مراحلش رو توضیح بدم. شاید پست وبلاگ چندان گزینه مناسبی نباشه. 

به هر حال پیام شما در inbox گیت‌هابم هست و پاکش نمی‌کنم تا انجامش بدم.

**Comment by fardm** [comment]
سلام امیر جان
ممنون از مطلبی که نوشتی
منم مدتیه از ابسیدین استفاده می کنم، با توضیحاتی که نوشتی ترغیب شدم و یه سایت استاتیک با هوگو راه انداختم.
فقط یه مشکلی دارم، لینک های داخلی ابسیدین توی هوگو کار نمیکنه نه Wikilink نه Markdown. من مجبورم از روش [ref](https://gohugo.io/content-management/cross-references/) استفاده کنم تا لینک ها کار کنه.
بعضی از تم ها رو نگاه کردم مثل [amethyst](https://github.com/64bitpandas/amethyst) و [quartz](https://github.com/jackyzha0/quartz) راحت لینک های ابسیدین رو رندر میکنه ولی سر درنیاوردم چه جوریه تا بتونم توی سایت خودم ازش استفاده کنم. (تم سایت من اینه:[stack](https://github.com/CaiJimmy/hugo-theme-stack))

می خواستم ببینم تو هم با این مشکل مواجه شدی؟ راه حلی براش پیدا کردی؟

**Reply by pourmand1376** [reply]
سلام. وقت بخیر. 

برای لینک‌ها من از قابلیت‌های هوگو استفاده نمی‌کنم. از ساده‌ترین شکل ممکن پیش میرم. مثلاً فرض کن نیاز دارم به صفحه [about](https://aprd.ir/about/) لینک بدم. 
```
[My About Page](https://aprd.ir/about/)
```
اینطور کدی رو می‌نویسم (همون‌طور که می‌دونی مارک‌داون استاندارد هست و چیز عجیبی نیست). به نظرم لزومی نداره که بصورت داینامیک با آدرس سایت لینک ساخته بشه. اگر هم زمانی - با احتمال خیلی کم‌ - قرار شد آدرس دامنه‌ سایتم عوض بشه، ته تهش Find and replace می‌کنم. 

البته یک پلاگین [Auto Link Title](https://github.com/zolrath/obsidian-auto-link-title) دارم که خیلی دوستش دارم. هر وقت لینکی رو Paste می‌کنم، خودش میره تایتل مناسب براش پیدا می‌کنه و می‌ذاره. 

مزیت این روش اینه که داخل Obsidian عادی نشون داده میشه. عیبش هم همونی هست که گفتم. اگر زمانی دامنه عوض بشه، قدری سخت‌تر خواهد بود.

**Reply by fardm** [reply]
آره این روش خیلی بهتره، ولی بدیش اینه که  بعضی قابلیت های ابسیدین مثل گراف ویو دیگه قابل استفاده نیست. (البته برای کسی که بخواد یه سایت عادی با ابسیدین و هوگو بسازه مشکلی نداره ولی من چون میخواستم تقریبا از روش [دیجیتال گاردن](https://maggieappleton.com/garden-history) استفاده کنم ویکی لینک و گراف ویو توی کارم زیاد استفاده میشه)

من الان از [obsidian-to-hugo](https://github.com/devidw/obsidian-to-hugo) استفاده میکنم که خودش لینک های ابسیدین رو به ref تبدیل میکنه. اینجوری یه مخزن اصلی دارم که روی اون کار میکنم هر موقع خواستم تغییرات رو منتشر کنم قبلش از این استفاده می کنم که لینک ها رو تبدیل کنه. هرچند خیلی سریع و ساده انجام میشه ولی خب یه مرحله اضافیه دیگه. 

روشی که تم quartz و amethyst استفاده میکنن خیلی راحت تره. من دوست داشتم بتونم این روش رو پیاده کنم ولی چون خیلی با این فضاها آشنا نیستم نتونستم بفهمم چه جوری کار میکنه.

**Comment by alidalvand** [comment]
کماکان منتظریم :)

**Reply by pourmand1376** [reply]
بفرمایید :) 

[آموزش ساخت وب‌سایت شخصی استاتیک - قسمت دوم | گاه‌نوشته‌های امیرپورمند](https://aprd.ir/create-your-own-website-part2/)
